misc/python/materialize/parallel_benchmark/scenarios.py

```diff
@@ -0,0 +1,1064 @@
+                    CREATE DATABASE public;
+                    USE public;
+                    CREATE TABLE t3 (f1 INTEGER);
+
+                    > CREATE SOURCE mysql_source
+                      FROM MYSQL CONNECTION mysql_conn
+                      FOR TABLES (public.t3);
+
+                    > CREATE MATERIALIZED VIEW mv_sum_mysql AS
+                      SELECT COUNT(*) FROM t3;
+
+                    > CREATE DEFAULT INDEX ON mv_sum_mysql;
+                    """
+                ),
+                LoadPhase(
+                    duration=120,
+                    actions=[
+                        OpenLoop(
+                            action=TdAction(
+                                f"""
+                                $ mysql-connect name=mysql url=mysql://root@mysql password={MySql.DEFAULT_ROOT_PASSWORD}
+                                $ mysql-execute name=mysql
+                                USE public;
+                                {"INSERT INTO t3 VALUES (1); " * 100}
+                                """,
+                                c,
+                            ),
+                            dist=Periodic(per_second=1),
+                        )
+                    ]
+                    + [
+                        ClosedLoop(
+                            action=StandaloneQuery(
+                                "SELECT * FROM mv_sum_mysql",
+                                conn_info=conn_infos["materialized"],
+                                strict_serializable=False,
+                            ),
+                        )
+                        for i in range(10)
+                    ],
+                ),
+            ],
+            guarantees={
+                "SELECT * FROM mv_sum_mysql (standalone)": {"qps": 15, "p99": 400},
+            },
+        )
+
+
+class OpenIndexedSelects(Scenario):
+    def __init__(self, c: Composition, conn_infos: dict[str, PgConnInfo]):
+        self.init(
+            [
+                TdPhase(
+                    """
+                    > CREATE TABLE t4 (f1 TEXT, f2 INTEGER);
+                    > CREATE DEFAULT INDEX ON t4;
+                    > INSERT INTO t4 VALUES ('A', 1);
+                    > INSERT INTO t4 VALUES ('B', 2);
+                    > INSERT INTO t4 VALUES ('C', 3);
+                    > INSERT INTO t4 VALUES ('D', 4);
+                    """
+                ),
+                LoadPhase(
+                    duration=120,
+                    actions=[
+                        OpenLoop(
+                            action=PooledQuery(
+                                "SELECT * FROM t4", conn_info=conn_infos["materialized"]
+                            ),
+                            dist=Periodic(per_second=400),
+                        ),
+                    ],
+                ),
+            ],
+            conn_pool_size=100,
+            guarantees={
+                "SELECT * FROM t4 (pooled)": {"qps": 200, "p99": 100},
```

Comment:
Shouldn't the guarantee be 400 in this case? If we're driving it at 400 QPS but only expect 200 QPS, then we're measuring something I don't understand!

Reply:
I was (too) conservative with these numbers, and put them at half of what I was seeing so that they wouldn't regress easily in CI for now. Will fix.